### PR TITLE
0001 john.curtis s3 example

### PR DESCRIPTION
**Pull request by JohnDCurtis**
This is the introductory S3 code.  We are already planning on enhancing it in various ways, as an FYI,

Please disregard the html file.  I will remove it.

### CODE REVIEW

File: SAIFENSDemo/java/compile.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java/compile.sh]: I will be removing this file.  no need to review it.

File: SAIFENSDemo/java/index.html [hunk 2]
```diff
@@ -0,0 +1,1135 @@
+<!--
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java/index.html]: I will be removing this file, no need to review it.

File: SAIFENSDemo/java/src/com/saife/sample/S3Sample.java [hunk 3]
```diff
@@ -0,0 +1,711 @@
...
+      } catch (final IOException e) {
+        System.out.println("CreateNetworkShare IOException");
+        e.printStackTrace();
+        return;
+      } catch (final NetworkShareExistsException e) {
+        System.out.println("Unrecoverable NetworkShareExistsException, since GetNetworkShare also failed.");
+        return;
+      }
+    }
+
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java/src/com/saife/sample/S3Sample.java]: In some cases the code will require this:    
try {
      workingObj.writeOut();
    } catch (final Exception e) {
      // do nothing
    }

File: SAIFENSDemo/java/src/com/saife/sample/S3Sample.java [hunk 4]
```diff
@@ -0,0 +1,711 @@
...
+    if (null == new_os) {
+      System.out.println("getEncryptStream returns null without a throw."); // should be impossible
+      return;
+    }
+
+    createSampleFile(new_os); // closes new_os
+    newObj.writeOut(); // saves the encrypted data out to S3
+
+    InputStream new_is = null;
+    System.out.println("Creating an encrypted file");
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java/src/com/saife/sample/S3Sample.java]: change to:
  System.out.println("Retrieving an decrypting a file");

File: SAIFENSDemo/java/src/com/saife/sample/S3Sample.java [hunk 5]
```diff
@@ -0,0 +1,711 @@
...
+    }
+
+  }
+
+  /**
+   * 
+   */
+  private void initS3() {
+    // S3 credential identity
+    final String me = "john.curtis@saife-tiprnet";
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java/src/com/saife/sample/S3Sample.java]: Replace with generic credentials.

File: SAIFENSDemo/java/src/com/saife/sample/S3Sample.java [hunk 6]
```diff
@@ -0,0 +1,711 @@
...
+
+    AWSCredentials credentials = null;
+    try {
+      credentials = new ProfileCredentialsProvider(me).getCredentials();
+    } catch (final Exception e) {
+      throw new AmazonClientException("Cannot load the credentials from the credential profiles file. "
+          + "Please make sure that your credentials file is at the correct "
+          + "location (/home/builder/.aws/credentials), and is in valid format.", e);
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java/src/com/saife/sample/S3Sample.java]: location is user specific.  remove.

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java [hunk 7]
```diff
@@ -0,0 +1,622 @@
...
+      blackDataHandler.sync();
+    } catch (final NoSuchContactException e) {
+      return false;
+    } catch (final InvalidManagementStateException e) {
+      return false;
+    } catch (final NotAllowedException e) {
+      // addMemeber not allowed
+      return false;
+    } catch (final UnlockRequiredException e) {
+      // addMember while locked
```
Comment (dickjones729) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: remove member

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java [hunk 8]
```diff
@@ -0,0 +1,622 @@
...
+
+    /** The objects. */
+    List<AnObject> objects;
+
+    /**
+     * The constructor.
+     * 
+     * @param s3m the S3 manager
+     */
+    public Persister(final S3Manager s3m) {
```
Comment (dickjones729) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: are you doing something with the s3m?

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java [hunk 9]
```diff
@@ -0,0 +1,622 @@
...
+        is.close();
+      } catch (final IOException e) {
+        System.out.println("NOTE: releaseInputStream ignored an IOException");
+      }
+    }
+
+    @Override
+    public OutputStream getOutputStream(final PersistedObject object) throws IOException {
+
+      // libsaife callbacks will return PersistedObjects, not the expanded AnObject class we defined
```
Comment (dickjones729) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: I thought that objects used to call getInput/OutputStream  should be the objects the persister returned via getObjects

Comment (dickjones729) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: You should just be able to cast here as the object should be an instanceof AnObject (and you might want to check instanceof before casting.  Here is an excerpt from an example Tony wrote.

  public OutputStream getOutputStream(final PersistedObject po) throws IOException {
    logger.trace("Enter: getOutputStream(PersistedObject)");
    if ((po instanceof FilePersistedObject) == false) {
      throw new IOException("The object can't be found");
    }

    final FilePersistedObject fpo = (FilePersistedObject) po;

    return fpo.getOutputStream();
  }

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java (hunk 8 above)
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: Not any more.  I removed that.

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java [hunk 10]
```diff
@@ -0,0 +1,622 @@
...
+        System.out.println("CreateNetworkShare IOException");
+        e.printStackTrace();
+        return;
+      } catch (final NetworkShareExistsException e) {
+        System.out.println("Unrecoverable NetworkShareExistsException, since GetNetworkShare also failed.");
+        return;
+      }
+    }
+
+    blackDataHandler.sync();
```
Comment (dickjones729) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: This should go away after fixing ticket 7661

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java (hunk 9 above)
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: Yeah, I forgot to revisit this after the library update.  It is now re-written this according to the example.  Also, I added an IOException throw.

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java (hunk 7 above)
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: yes, copypasta error.  i removed it.

File: SAIFENSDemo/java_swing/src/com/saife/sample/MainFrameLauncher.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
+/*
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/MainFrameLauncher.java]: GLOBAL: Replaced copyright with Apache License

File: SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java (hunk 10 above)
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/SaifeManager.java]: I added this tag everywhere sync is called:
// Should not be needed.  Refer to ticket 7661.

File: SAIFENSDemo/java/src/com/saife/sample/S3Sample.java [hunk 12]
```diff
@@ -0,0 +1,711 @@
+/*
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java/src/com/saife/sample/S3Sample.java]: In all likelihood we will remove this code.

File: SAIFENSDemo/java_swing/src/com/saife/sample/NSSettingsFrame.java [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+
+/**
+ * The NSSettingsFrame.
+ */
+public class NSSettingsFrame {
+
+  /** The networkShareSettings. */
+  private JFrame networkShareSettings;
+
+  /** The parent. */
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/NSSettingsFrame.java]: I removed parent.  it is unused.

File: SAIFENSDemo/java_swing/src/com/saife/sample/NSWorkingFrame.java [hunk 14]
```diff
@@ -0,0 +1,236 @@
...
+import java.util.List;
+
+import javax.swing.DefaultListModel;
+import javax.swing.JButton;
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JList;
+
+public class NSWorkingFrame {
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/NSWorkingFrame.java]: added java doc to class and variables.

File: SAIFENSDemo/java_swing/src/com/saife/sample/NSWorkingFrame.java [hunk 15]
```diff
@@ -0,0 +1,236 @@
...
+    final String sel = (String) networkShareList.getSelectedValue();
+    lm.removeElement(sel);
+  }
+
+  /**
+   * Leave this NetworkShare working dialogue
+   */
+  void handleBack() {
+    System.out.println("go back");
+    if (null != parent) {
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/NSWorkingFrame.java]: This check is no longer needed.  i deleted it.

File: SAIFENSDemo/java_swing/src/com/saife/sample/NSWorkingFrame.java [hunk 16]
```diff
@@ -0,0 +1,236 @@
...
+import javax.swing.JLabel;
+import javax.swing.JList;
+
+public class NSWorkingFrame {
+
+  private JFrame netShareWorkingFrame;
+
+  public String nsName;
+
+  MainFrame parent;
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/NSWorkingFrame.java]: removed parent and all references.  this was no longer needed.

File: SAIFENSDemo/java_swing/src/com/saife/sample/NameNetworkShare.java [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+import java.awt.event.ActionListener;
+
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import javax.swing.border.EmptyBorder;
+
+public class NameNetworkShare extends JFrame {
```
Comment (JohnDCurtis) [inline SAIFENSDemo/java_swing/src/com/saife/sample/NameNetworkShare.java]: added java doc for class and all variables.

**Pull request merged**